tests/src/performance/Scenario/JitBench/JitBenchHarness.cs

```diff
@@ -0,0 +1,489 @@
+            using (var client = new HttpClient())
+            {
+                var archiveName = $"{JitBenchCommitSha1Id}.zip";
+                var url = $"{JitBenchRepoUrl}/archive/{archiveName}";
+                var zipFile = Path.Combine(s_temporaryDirectory, archiveName);
+
+                using (FileStream tmpzip = File.Create(zipFile))
+                {
+                    using (Stream stream = client.GetStreamAsync(url).Result)
+                        stream.CopyTo(tmpzip);
+                    tmpzip.Flush();
+                }
+
+                // If the repo already exists, we delete it and extract it again.
+                if (Directory.Exists(s_jitBenchDevDirectory))
+                    Directory.Delete(s_jitBenchDevDirectory, true);
+
+                // This step will create s_JitBenchDevDirectory.
+                ZipFile.ExtractToDirectory(zipFile, s_temporaryDirectory);
+                Directory.Move(Path.Combine(s_temporaryDirectory, $"JitBench-{JitBenchCommitSha1Id}"), s_jitBenchDevDirectory);
+            }
+        }
+
+        private static void InstallSharedRuntime()
+        {
+            var psi = new ProcessStartInfo() {
+                WorkingDirectory = s_jitBenchDevDirectory,
+                FileName = @"powershell.exe",
+                Arguments = $".\\Dotnet-Install.ps1 -SharedRuntime -InstallDir .dotnet -Channel master -Architecture {s_targetArchitecture}"
+            };
+            LaunchProcess(psi, 180000);
+        }
+
+        private static IDictionary<string, string> InstallDotnet()
+        {
+            var psi = new ProcessStartInfo() {
+                WorkingDirectory = s_jitBenchDevDirectory,
+                FileName = @"powershell.exe",
+                Arguments = $".\\Dotnet-Install.ps1 -InstallDir .dotnet -Channel master -Architecture {s_targetArchitecture}"
+            };
+            LaunchProcess(psi, 180000);
+
+            // TODO: This is currently hardcoded, but we could probably pull it from the powershell cmdlet call.
+            return new Dictionary<string, string> { { "PATH", $"{Path.Combine(s_jitBenchDevDirectory, ".dotnet")};{psi.Environment["PATH"]}" } };
+        }
+
+        private static void ModifySharedFramework()
+        {
+            // Current working directory is the <coreclr repo root>/sandbox directory.
+            Console.WriteLine($"Modifying the shared framework.");
+
+            var sourcedi = new DirectoryInfo(Directory.GetCurrentDirectory());
+            var targetdi = new DirectoryInfo(
+                new DirectoryInfo(Path.Combine(s_jitBenchDevDirectory, ".dotnet", "shared", "Microsoft.NETCore.App"))
+                .GetDirectories("*")
+                .OrderBy(s => s.Name)
+                .Last()
+                .FullName);
+
+            Console.WriteLine($"  Source : {sourcedi.FullName}");
+            Console.WriteLine($"  Target : {targetdi.FullName}");
+
+            var compiledBinariesOfInterest = new string[] {
```

Comment:
We're doing a patch of the compiled runtime here, which is something that can be fragile going forward.  We should surface this as an issue with the eng team and ask what the right way to do this is - perhaps a full-stack build.